code/game/objects/structures/door_assembly.dm

```diff
@@ -579,63 +583,63 @@
 			if(C.get_amount() < 1 || state != 0)
 				return
 			C.use(1)
-			src.state = 1
+			state = AIRLOCK_ASSEMBLY_NEEDS_ELECTRONICS
 			to_chat(user, "<span class='notice'>You wire the airlock assembly.</span>")
-			src.name = "wired airlock assembly"
+			name = "wired airlock assembly"
 
 	else if(istype(W, /obj/item/wirecutters) && state == 1 )
-		playsound(src.loc, W.usesound, 100, 1)
+		playsound(src, W.usesound, 100, 1)
 		user.visible_message("[user] cuts the wires from the airlock assembly.", \
 							"<span class='notice'>You start to cut the wires from the airlock assembly...</span>")
 
 		if(do_after(user, 40*W.toolspeed, target = src))
-			if( src.state != 1 )
+			if(state != 1)
 				return
 			to_chat(user, "<span class='notice'>You cut the wires from the airlock assembly.</span>")
 			new/obj/item/stack/cable_coil(get_turf(user), 1)
-			src.state = 0
-			src.name = "secured airlock assembly"
+			state = AIRLOCK_ASSEMBLY_NEEDS_WIRES
+			name = "secured airlock assembly"
 
 	else if(istype(W, /obj/item/electronics/airlock) && state == 1 )
-		playsound(src.loc, W.usesound, 100, 1)
+		playsound(src, W.usesound, 100, 1)
 		user.visible_message("[user] installs the electronics into the airlock assembly.", \
 							"<span class='notice'>You start to install electronics into the airlock assembly...</span>")
 		if(do_after(user, 40, target = src))
-			if( src.state != 1 )
+			if( state != 1 )
 				return
 			if(!user.transferItemToLoc(W, src))
 				return
 
 			to_chat(user, "<span class='notice'>You install the airlock electronics.</span>")
-			src.state = 2
-			src.name = "near finished airlock assembly"
-			src.electronics = W
+			state = AIRLOCK_ASSEMBLY_NEEDS_SCREWDRIVER
+			name = "near finished airlock assembly"
+			electronics = W
 
 
 	else if(istype(W, /obj/item/crowbar) && state == 2 )
-		playsound(src.loc, W.usesound, 100, 1)
+		playsound(src, W.usesound, 100, 1)
 		user.visible_message("[user] removes the electronics from the airlock assembly.", \
 								"<span class='notice'>You start to remove electronics from the airlock assembly...</span>")
 
 		if(do_after(user, 40*W.toolspeed, target = src))
-			if( src.state != 2 )
+			if(state != 2)
 				return
 			to_chat(user, "<span class='notice'>You remove the airlock electronics.</span>")
-			src.state = 1
-			src.name = "wired airlock assembly"
+			state = AIRLOCK_ASSEMBLY_NEEDS_ELECTRONICS
+			name = "wired airlock assembly"
 			var/obj/item/electronics/airlock/ae
 			if (!electronics)
-				ae = new/obj/item/electronics/airlock( src.loc )
+				ae = new/obj/item/electronics/airlock( loc )
 			else
 				ae = electronics
 				electronics = null
-				ae.loc = src.loc
+				ae.forceMove(src)
```

Comment:
shouldn't you move this to src.loc still? before this was being moved to the source objects location, now it's being moved into the source objects contents

maybe you want ae.forcMove(get_tile(src))